flink-yarn/src/main/java/org/apache/flink/yarn/ApplicationReportProviderImpl.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.yarn;
+
+import org.apache.flink.annotation.Internal;
+
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+import org.apache.hadoop.yarn.api.records.ApplicationReport;
+import org.apache.hadoop.yarn.api.records.YarnApplicationState;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+
+import java.io.IOException;
+
+/** Then implementation of ApplicationReportProvider. */
+@Internal
+public class ApplicationReportProviderImpl implements ApplicationReportProvider {
+    private YarnClient yarnClient;
+    private ApplicationId appId;
+
+    private ApplicationReportProviderImpl(YarnClient yarnClient, ApplicationId applicationId) {
+        this.yarnClient = yarnClient;
+        this.appId = applicationId;
+    }
+
+    @Override
+    public ApplicationReport waitTillSubmissionFinish() throws Exception {
+        try {
+            return YarnClusterDescriptor.waitTillTargetState(
+                    yarnClient, appId, YarnApplicationState.RUNNING);
+        } catch (IOException e) {
+            throw new RuntimeException(
+                    "Errors on getting YARN application report. Maybe application has finished.",
+                    e);
```

Comment:
Why throwing an unchecked exception when the signature states a checked exception?